include/pmacc/math/vector/Vector.hpp

```diff
@@ -23,99 +23,47 @@
 
 #include <boost/call_traits.hpp>
 
+#include <array>
 #include <iostream>
 #include <type_traits>
 
 namespace pmacc
 {
     namespace math
     {
-        namespace detail
+        template<typename T_Type, uint32_t T_dim>
+        struct ArrayStorage : private std::array<T_Type, T_dim>
         {
-            template<typename T_Type, uint32_t T_dim>
-            struct Vector_components
-            {
-                static constexpr bool isConst = false;
-                static constexpr uint32_t dim = T_dim;
-                using type = T_Type;
-
-                constexpr Vector_components() = default;
-
-                constexpr Vector_components& operator=(const Vector_components&) = default;
-
-                constexpr Vector_components(const Vector_components&) = default;
-
-                template<
-                    typename T_Navigator,
-                    typename... T_Args,
-                    typename = std::enable_if_t<
-                        (std::is_convertible_v<T_Args, T_Type> && ...)
-                        && !std::is_same_v<T_Navigator, IdentityNavigator>>>
-                HDINLINE constexpr Vector_components(T_Navigator const& navigator, T_Args... args)
-                {
-                    static_assert(
-                        sizeof...(T_Args) == dim,
-                        "Number of arguments must be equal to the vector dimension.");
-                    int i = 0;
-                    ([&] { operator[](navigator(i++)) = static_cast<type>(args); }(), ...);
-                }
-
-                template<
-                    typename... T_Args,
-                    typename = std::enable_if_t<(std::is_convertible_v<T_Args, T_Type> && ...)>>
-                HDINLINE constexpr Vector_components(IdentityNavigator const&, T_Args... args)
-                    : v{static_cast<type>(args)...}
-                {
-                    static_assert(
-                        sizeof...(T_Args) == dim,
-                        "Number of arguments must be equal to the vector dimension.");
-                }
-
-                /*align full vector*/
-                PMACC_ALIGN(v[dim], type) = {};
-
-                HDINLINE
-                constexpr type& operator[](const uint32_t idx)
-                {
-                    return v[idx];
-                }
+            using BaseType = std::array<T_Type, T_dim>;
+            // using BaseType::array;
+            using BaseType::operator[];
 
-                HDINLINE
-                const type& operator[](const uint32_t idx) const
-                {
-                    return v[idx];
-                }
-            };
+            constexpr ArrayStorage() = default;
+            constexpr ArrayStorage(ArrayStorage const&) = default;
+            constexpr ArrayStorage(ArrayStorage&&) = default;
+            constexpr ArrayStorage& operator=(ArrayStorage const&) = default;
+            constexpr ArrayStorage& operator=(ArrayStorage&&) = default;
```

Comment:
``` suggestion
```
These are all defaulted. This is correct, but they can simply be removed and the compiler generates the same stuff.